tool/src/main/java/com/fidesmo/fdsm/Main.java

```diff
@@ -262,7 +262,7 @@ public static void main(String[] argv) {
                 Card card = terminal.connect("*");
                 Optional<byte[]> uid = getUID(card.getBasicChannel());
```

Comment:
As I see this UID is only used for batching URL, so I would rather move it to the `FidesmoCard` (and reading as well). That will be more convenient to use and won't break encapsulation.

Reply:
I did it that way but then I decided to change and reuse that getUID command as it's the very first command that needs to be sent (previous to any other SELECT). Let me check if the probe could go after any other APDU exchange (which shouldn't) and it that case I'll move UID back to `FidesmoCard`